src/algorithms/extractor/musicextractor.cpp

```diff
@@ -585,5 +610,42 @@ void MusicExtractor::setExtractorOptions(const std::string& filename) {
   options.merge(opts, "replace");
 }
 
+#if HAVE_LIBCHROMAPRINT
+void MusicExtractor::computeChromaPrint(const string& audioFilename, Pool& results) {
+  AlgorithmFactory& factory = standard::AlgorithmFactory::instance();
+
+  Algorithm* audio = factory.create("MonoLoader",
+                                    "filename", audioFilename,
+                                    "sampleRate", analysisSampleRate,
+                                    "downmix", downmix);
+  Algorithm* chromaprinter = factory.create("Chromaprinter",
+                                            "sampleRate", analysisSampleRate,
+                                            "maxLength", chromaprintDuration);
+
+  vector<Real> siganl;
+  string chromaprint;
+
+  audio->output("audio").set(siganl);
+  chromaprinter->input("signal").set(siganl);
+
+  chromaprinter->output("fingerprint").set(chromaprint);
+
+  try {
+    audio->compute();
+    chromaprinter->compute();
+
+    results.add("chromaprint.string", chromaprint);
+    if (chromaprintDuration == 0.f)
+      results.add("chromaprint.duration", results.value<Real>("metadata.audio_properties.length"));
+    else
+    results.add("chromaprint.duration", chromaprintDuration);
+  }
+  catch (const EssentiaException&) {
+    throw EssentiaException("File looks like a completely silent file");
```

Comment:
This is not informative. Use the text from the original exception (thrown by MonoLoader or Chromaprinter) instead.